backend/data_preparation/dumper/fire_dumper.py

```diff
@@ -0,0 +1,438 @@
+    def retrieve_all_fires() -> Set[FireEvent]:
+        """
+        Retrieves all fires in the database.
+        :return: set of FireEvent objects
+                 e.g. [FireEvent(-1, 2015, 'California', 'FireA'), FireEvent(-1, 2015, 'California', 'FireB')]
+        """
+        # check if the fire_history table exists
+        # if not exist, executing sql_retrieve_all_fires will return an error
+        FireDumper._create_table("fire_history")
+        # retrieve all fires in fire_history
+        set_of_fire_event_objects = list(map(lambda fire_event_tuple: FireEvent.from_tuple(fire_event_tuple),
+                                        Connection.sql_execute(FireDumper.SQL_RETRIEVE_ALL_FIRES)))
+        # result now is a set of FireEvent objects
+        # e.g. for https://rmgsc.cr.usgs.gov/outgoing/GeoMAC/2015_fire_data/California/Deer_Horn_2/
+        # the FireEvent object is: Fire Event: Deer_Horn_2 in year 2015, state California
+        return set_of_fire_event_objects
+
+    @staticmethod
+    def _generate_sql_statement_and_execute(sql_statement: str, data: Dict[str, Any]):
+        """
+        Generates a SQL statement with the data given as a dictionary and execute, commit the changes.
+        :param sql_statement: string
+                e.g. "SELECT * FROM fire_history WHERE %{id}=199"
+        :param data: data as a dict.
+                e.g. {'year': 2019, 'firename': 'TRESTLE', 'agency': 'USFS', 'datetime': .....}
+        """
+        with Connection() as conn:
+            cur = conn.cursor()
+            cur.execute(sql_statement, data)
+            conn.commit()
+            cur.close()
+        return
+
+    def insert(self, data: Dict[str, Any]):
+        """
+        Inserts a single fire record from FireExtractor.extract() into fire table.
+        :param data:dict to insert, from extractor.extract()
+                e.g. {'year': 2019, 'firename': 'TRESTLE', 'agency': 'USFS', 'datetime': .....}
+        """
+        logger.info(f"Inserting into fire table: {data['firename']} {data['datetime']}")
+        # check if the fire table exists
+        # if not exist, executing sql_insert will cause an error
+        FireDumper._create_table("fire")
+        FireDumper._generate_sql_statement_and_execute(FireDumper.SQL_INSERT_FIRE, data)
+        logger.info(f"Finished inserting file: {data['firename']}{data['datetime']}")
+
+    @staticmethod
+    def insert_history(fire: FireEvent) -> None:
+        """
+        Inserts a fire record into fire_history table
+        :param fire: a FireEvent object representing a wild fire event
+                e.g. FireEvent(-1, 2015, 'California', 'FireQ')
+        :return: None
+        """
+        logger.info(f"Inserting into fire_history table: {fire.url_name} in {fire.state} in {fire.year}")
+        # make the dictionary to be inserted into fire_history
+        FireDumper._generate_sql_statement_and_execute(FireDumper.SQL_INSERT_FIRE_HISTORY, fire.to_dict())
```

Comment:
no need to ensure `fire_history` table is existing? (like the logic in `insert()`)